functions/Resolve-DbaNetworkName.ps1

```diff
@@ -93,175 +87,127 @@ Function Resolve-DbaNetworkName
+				Write-Message -Level Verbose -Message "Your PowerShell Version is $($host.Version.Major)"
+				try {
+					Write-Message -Level Verbose -Message "Getting computer information from $Computer"
+					if (Was-Bound "Credential") {
+						$conn = Get-DbaCmObject -ClassName win32_ComputerSystem -Computer $Computer -Credential $Credential
 					}
-					else
-					{
-						$conn = Get-CimInstance -Query "Select * FROM Win32_computersystem" -ComputerName $Computer -ErrorAction Stop
+					else {
+						$conn = Get-DbaCmObject -ClassName win32_ComputerSystem -Computer $Computer
 					}
 				}
-				catch
-				{
-					Write-Verbose "$functionName - No WSMan connection to $Computer"
-				}
-				if (!$conn)
-				{
-					try
-					{
-						Write-Verbose "$functionName - Getting computer information from $Computer via CIM (DCOM)"
-						$sessionoption = New-CimSessionOption -Protocol DCOM
-						if ($Credential)
-						{
-							$CIMsession = New-CimSession -ComputerName $Computer -SessionOption $sessionoption -ErrorAction Stop -Credential $Credential
-							
-						}
-						else
-						{
-							$CIMsession = New-CimSession -ComputerName $Computer -SessionOption $sessionoption -ErrorAction Stop
-						}
-						
-						$conn = Get-CimInstance -Query "Select * FROM Win32_computersystem" -CimSession $CIMsession -ErrorAction Stop
-					}
-					catch
-					{
-						Write-Warning "$functionName - No DCOM connection for CIM to $Computer"
-					}
+				catch {
+					Write-Message -Level Verbose -Message "Unable to get computer information from $Computer"
 				}
-				
-				if (!$conn)
-				{
-					Write-Verbose "$functionName - No CIM from $Computer. Getting HostName via .NET.Dns"
-					try
-					{
+
+				if (!$conn) {
+					Write-Message -Level Verbose -Message "No WMI/CIM from $Computer. Getting HostName via .NET.Dns"
+					try {
 						$fqdn = ([System.Net.Dns]::GetHostEntry($Computer)).HostName
 						$hostname = $fqdn.Split(".")[0]
-						
+
 						$conn = [PSCustomObject]@{
-							Name = $Computer
+							Name        = $Computer
 							DNSHostname = $hostname
-							Domain = $fqdn.Replace("$hostname.", "")
+							Domain      = $fqdn.Replace("$hostname.", "")
 						}
 					}
-					catch
-					{
-						Write-Warning "$functionName - No .NET.Dns information from $Computer"
+					catch {
+						Write-Message -Level Warning -Message "No .NET.Dns information from $Computer"
 						continue
 					}
 				}
 			}
-			
-			
-			try
-			{
-				Write-Verbose "$functionName - Resolving $($conn.DNSHostname) using .NET.Dns GetHostEntry"
+
+			try {
+				Write-Message -Level Verbose -Message "Resolving $($conn.DNSHostname) using .NET.Dns GetHostEntry"
 				$hostentry = ([System.Net.Dns]::GetHostEntry($conn.DNSHostname)).HostName
 			}
-			catch
-			{
-				Write-Warning "$functionName - .NET.Dns GetHostEntry failed for $($conn.DNSHostname)"
+			catch {
+				Write-Message -Level Warning -Message ".NET.Dns GetHostEntry failed for $($conn.DNSHostname)"
```

Comment:
Should we continue, if this fails?
If not: `Stop-Function`
Also please pass along exceptions